### PR TITLE
Logger, Exception Handling, Design Change: Return Old Object

### PR DESCRIPTION
**Pull request by RathoreDevansh08**
Fixes convict-git/KoraKaagaz#93

### CODE REVIEW

**Comment by RathoreDevansh08** [comment]
@sakshi1603 @wildcat26 PR is ready for review. Indentation of 4 spaces is followed. It appears un-indented because GitHub's editor has a different default indentation width.

File: src/processing/ParameterizedOperationsUtil.java [hunk 1]
```diff
@@ -3,26 +3,43 @@
 import java.util.*;
 import processing.boardobject.*;
 import processing.utility.*;
+import infrastructure.Logger.LoggerFactory;
```
Comment (tymefighter) [inline src/processing/ParameterizedOperationsUtil.java]: @RathoreDevansh08 I think the logger is in `infrastructure.validation.logger.ILogger` and not here

Comment (RathoreDevansh08) [reply src/processing/ParameterizedOperationsUtil.java]: I'll check and change it

**Comment by RathoreDevansh08** [comment]
@sakshi1603 @wildcat26 PR is ready for review. Indentation of 4 spaces is followed. It appears un-indented because GitHub's editor has a different default indentation width.

File: src/processing/ParameterizedOperationsUtil.java [hunk 2]
```diff
@@ -3,26 +3,40 @@
...
 
+
 /**
  * Class for implementing selected(non-erase, non-reset) object's Color-change and Rotate
  * operation.
  */
 public class ParameterizedOperationsUtil {
 
+    // initialise a reference to the logger singleton object
+	ILogger paraOpLogger = LoggerFactory.getLogger();
```
Comment (wildcat26) [inline src/processing/ParameterizedOperationsUtil.java]: Logger Factory does not have a `getLogger`method. Its `getLoggerInstance`.

Comment (RathoreDevansh08) [reply src/processing/ParameterizedOperationsUtil.java]: done. thanks.

File: src/bin/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/bin/
```
Comment (sakshi1603) [inline src/bin/.gitignore:1]: Please remove this from your git

File: src/processing/ParameterizedOperationsUtil.java [hunk 4]
```diff
@@ -3,26 +3,40 @@
...
+
+			UndoRedo.pushIntoStack(newObj);
+
+			paraOpLogger.log(moduleID.PROCESSING, logLevelID.SUCCESS, "Undo-Redo stacks updated");
+		} catch (Exception e) {
+
+            paraOpLogger.log(moduleID.PROCESSING, logLevelID.ERROR, "Undo-Redo call failed!");
+        }
+
         return;
```
Comment (sakshi1603) [inline src/processing/ParameterizedOperationsUtil.java:40]: Please check the indentation.

File: src/processing/ParameterizedOperationsUtil.java [hunk 5]
```diff
@@ -3,26 +3,40 @@
...
 public class ParameterizedOperationsUtil {
 
+    // initialise a reference to the logger singleton object
+	ILogger paraOpLogger = LoggerFactory.getLoggerInstance();
+
     /**
      * Function to update undo-redo stack. Undo-Redo module cannot call this function inorder to
      * avoid logical errors.
      */
     private static void stackUtil(BoardObject newObj) {
```
Comment (sakshi1603) [inline src/processing/ParameterizedOperationsUtil.java:28]: Sir asked to change the name of stackUtil to Satchit. Can you also change it.

**Pull request closed without merging**